services/security_group_manager/src/main/java/com/futurewei/alcor/securitygroup/exception/DefaultSecurityGroupExists.java

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.securitygroup.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(code= HttpStatus.PRECONDITION_FAILED, reason="Default security group exists")
+public class DefaultSecurityGroupExists extends Exception {
```

Comment:
I think we should have a good design doc and include details of all these good exception.

Reply:
no problem